mssqlcli/mssql_cli.py

```diff
@@ -187,31 +183,22 @@ def __init__(self, options):
             self.completer = MssqlCompleter(smart_completion=smart_completion, settings=self.settings)
             self._completer_lock = threading.Lock()
 
+        # output file is for non-interactive mode
+        self.output_file = options.output_file
+
         self.sqltoolsclient = SqlToolsClient(enable_logging=options.enable_sqltoolsservice_logging)
         self.mssqlcliclient_main = MssqlCliClient(options, self.sqltoolsclient)
 
+        # exit and return error if user enters interactive mode with -o argument enabled
+        if self.interactive_mode and self.output_file:
+            click.secho("Invalid request: -o must be accompanied with a query.", err=True, fg='red')
+            sys.exit(1)
+
     def __del__(self):
         # Shut-down sqltoolsservice
         if self.sqltoolsclient:
             self.sqltoolsclient.shutdown()
 
-    def write_to_file(self, pattern, **_):
```

Comment:
This doesn't get used by anything, but may eventually be used for \o special query. @pensivebrian--up to you to keep it or not.

Reply:
I'd leave it if the \o special command uses it.